pkg/controller/apischeme/apischeme_controller_test.go

```diff
@@ -0,0 +1,415 @@
+				AllowedCIDRBlocks: cidrs,
+			},
+		},
+	}
+}
+func TestMasterNodeSubnets(t *testing.T) {
+	tests := []struct {
+		region       string
+		az           string
+		clustername  string
+		cidrs        []string
+		endpointName string
+		masterCount  int
+		apiEndpoint  string
+
+		mockSubnetIDInput   []string // name to id lookup input
+		mockSubnetIDOutput0 []string // name to id lookup output
+		mockSubnetIDOutput1 error    // name to id lookup error
+	}{
+		{
+			region:       defaultRegionName,
+			az:           defaultAzName,
+			clustername:  "test1",
+			cidrs:        []string{"0.0.0.0/0"},
+			endpointName: "rh-api",
+			masterCount:  3,
+			apiEndpoint:  defaultAPIEndpoint,
+
+			mockSubnetIDInput:   []string{"test1-public-" + defaultAzName},
+			mockSubnetIDOutput0: []string{"subnet-12345"},
+			mockSubnetIDOutput1: nil,
+		},
+		{
+			region:       defaultRegionName,
+			az:           defaultAzName,
+			clustername:  "test2",
+			cidrs:        []string{"0.0.0.0/0"},
+			endpointName: "rh-api",
+			masterCount:  3,
+			apiEndpoint:  defaultAPIEndpoint,
+
+			mockSubnetIDInput:   []string{"test2-public-" + defaultAzName},
+			mockSubnetIDOutput0: []string{"subnet-abcdef"},
+			mockSubnetIDOutput1: nil,
+		},
+	}
+
+	for testNumber, test := range tests {
+		aObj := makeAPISchemeObj(test.endpointName, true, test.cidrs)
+		masterNames := make([]string, test.masterCount)
+		for i := 0; i < test.masterCount; i++ {
+			masterNames[i] = fmt.Sprintf("master-%d", i)
+		}
+		machineList, _ := createMachineObjectList(masterNames, test.clustername, "master", test.region, test.az)
+		infraObj := infraObj(test.clustername, test.apiEndpoint, test.apiEndpoint, test.region)
+		objs := []runtime.Object{aObj, infraObj, machineList}
+		mocks := setupTests(t, objs)
+
+		expectedPublicSubnetName := fmt.Sprintf("%s-public-%s", test.clustername, test.az)
+		res, err := utils.GetMasterNodeSubnets(mocks.fakeKubeClient)
+		if err != nil {
+			t.Fatalf("Test %d Couldn't get the master node subnets: %v", testNumber, err)
+		}
+		if res["public"] != expectedPublicSubnetName {
+			t.Fatalf("Subnet mismatch. Got %s, expected %s", res["public"], expectedPublicSubnetName)
+		}
+		s := []string{res["public"]}
+		// now safe to mock the name -> id lookup
+		mocks.mockAws.EXPECT().SubnetNameToSubnetIDLookup(test.mockSubnetIDInput).AnyTimes().Return(test.mockSubnetIDOutput0, test.mockSubnetIDOutput1)
+
+		subnets, err := mocks.mockAws.SubnetNameToSubnetIDLookup(s)
+		if err != nil {
+			t.Fatalf("Couldn't lookup subnet IDs from name: %v", err)
+		}
+
+		if len(subnets) != len(test.mockSubnetIDInput) {
+			t.Fatalf("Test %d Wrong number of subnets back. Got %d, expected %d", testNumber, len(subnets), 1)
+		}
+		for i := range subnets {
+			if subnets[i] != test.mockSubnetIDOutput0[i] {
+				t.Fatalf("Wrong subnet ID back for public. Got %s, expected %s", subnets[i], test.mockSubnetIDOutput0[i])
+			}
+		}
+	}
+}
+
+func TestClusterBaseDomain(t *testing.T) {
```

Comment:
since this is testing `utils.GetClusterBaseDomain()` does it make more sense for this to be tested in the utils package?

Reply:
Yep, it is in fact in those tests (PR forthcoming). I think once they're committed, I'll remove the test here (and the other utils-focused tests).